hack/build/ci/sanitize-branch-name.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+readonly BRANCH_NAME=${1}
+
+if [ -z "${BRANCH_NAME}" ]; then
+  echo "Usage: $0 <branch name>"
+  exit 1
+fi
+
+echo "${BRANCH_NAME//[^a-zA-Z0-9_-]/-}"
```

Comment:
can we just include the . in our valid characters so that we have parity with docker metadata?
```suggestion
echo "${BRANCH_NAME//[^a-zA-Z0-9_.-]/-}"
```